### PR TITLE
Site rearrangement

### PR DESCRIPTION
**Pull request by snazy**
* Simplifies navigation (replace 9 sections with 5)
* Moves pages "around"
* Moves Nessie release dependent files to separate directory
* Adds Nessie version dependent docs - released versions end in the `site-docs` branch in the `projectnessie/nessie` GitHub repo
* Adds helper scripts + Makefile to develop locally (`make build` + `make serve`)
* Adds helper scripts for the Nessie `release-create` workflow
* Use the `blog` plugin for the blog
* Use the `monorepo` plugin for version dependent docs
* Use the `privacy` plugin
* Use the `redirects` plugin to let existing links work (being redirected)

### CODE REVIEW

File: site/docs/blog/posts/2023-05-namespace-enforcement.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
+---
+date: 2023-05-08
+authors:
+  - abhat
+---
+
 # Namespace enforcement in Nessie
 
 Starting from Nessie version 0.52.3, it is required to have existing Namespaces before creating or committing to tables.  
+<!-- more -->
```
Comment (dimas-b) [inline site/docs/blog/posts/2023-05-namespace-enforcement.md:10]: What does this mean / do?

File: site/docs/blog/index.md [hunk 2]
```diff
@@ -1,7 +1,6 @@
...
+hide:
+  - navigation
+---
 
-Highlights
-
-* [Rolling upgrade issue to 0.26.0](incident-2022-05.md) (May 2022)
-* [Namespace enforcement](namespace-enforcement.md) (May 2023)
-* [Introducing Nessie as a Dremio Source](https://www.dremio.com/blog/introducing-nessie-as-a-dremio-source/) (June 2023)
+# Blog
```
Comment (dimas-b) [inline site/docs/blog/index.md:6]: Why not `Nessie Blog` (as before)?

**Review by dimas-b — COMMENTED** [review]
Nice work :+1:

File: site/mkdocs.yml [hunk 3]
```diff
@@ -60,6 +127,8 @@ markdown_extensions:
   - markdown.extensions.def_list
   - markdown.extensions.footnotes
   - meta
+  - toc:
+      permalink: 🔗
```
[the remark this answers is not in the document]
Comment (snazy) [reply site/mkdocs.yml:131]: yes (don't know exactly why tho)

File: site/in-dev/index-release.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+## Download options for this Nessie ::NESSIE_VERSION:: release
+
+### Nessie Server as Docker image
+
+Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
+
+=== "GitHub Container Registry"
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://ghcr.io/projectnessie/nessie)
+    ```bash
+    docker pull ghcr.io/projectnessie/nessie:::NESSIE_VERSION::
```
[the remark this answers is not in the document]
Comment (snazy) [reply site/in-dev/index-release.md:17]: oh, it's in mkdocs.yml

File: site/docs/blog/posts/2023-05-namespace-enforcement.md (hunk 1 above)
Comment (snazy) [reply site/docs/blog/posts/2023-05-namespace-enforcement.md:10]: https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#adding-an-excerpt

File: site/docs/blog/index.md (hunk 2 above)
Comment (snazy) [reply site/docs/blog/index.md:6]: fixed

**Review by dimas-b — APPROVED** [review]

File: site/mkdocs.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (snazy) [reply site/mkdocs.yml:115]: It's just not the default symbol

**Pull request merged**